cmd/kubeadm/app/cmd/phases/join/preflight.go

```diff
@@ -141,15 +143,17 @@ func runPreflight(c workflow.RunData) error {
 
 // checkIfReadyForAdditionalControlPlane ensures that the cluster is in a state that supports
 // joining an additional control plane instance and if the node is ready to preflight
-func checkIfReadyForAdditionalControlPlane(initConfiguration *kubeadmapi.ClusterConfiguration) error {
+func checkIfReadyForAdditionalControlPlane(initConfiguration *kubeadmapi.ClusterConfiguration, hasCertificateKey bool) error {
 	// blocks if the cluster was created without a stable control plane endpoint
 	if initConfiguration.ControlPlaneEndpoint == "" {
 		return errors.New("unable to add a new control plane instance a cluster that doesn't have a stable controlPlaneEndpoint address")
 	}
 
-	// checks if the certificates that must be equal across contolplane instances are provided
-	if ret, err := certs.SharedCertificateExists(initConfiguration); !ret {
-		return err
+	if !hasCertificateKey {
```

Comment:
IMO checking all the certificates is a good thing, no matter if they are provided by the user or downloaded.
What about removing `hasCertificateKey`?

Reply:
This is not possible because the preflight check that the certificates exist, and `kubeadm` errors out, because on the preflight phase the certificates hasn't been downloaded yet from the secret, and the join process fails.

Reply:
kk